Docs for Immersive Intelligence mod (1.12.2) (#503)
CraftTweaker/CraftTweaker-Documentation: docs/1.12/content/Mods/Immersive_Intelligence/CraftTweaker_Support/Ammunition_Workshop.md

::requiredMod[Immersive Intelligence]{builtIn=true
modLink=https://www.curseforge.com/minecraft/mc-mods/immersive-intelligence}

# Ammunition Workshop

> The `AmmunitionWorkshop` package can be used to add/remove recipes to/from the II Ammunition Workshop.

Reviewer: ```suggestion
The `AmmunitionWorkshop` package can be used to modify the Immersive Intelligence Ammunition Workshop recipes.
```

**Take note it is not limited to producing ammunition and can be used for any generic recipe involving 2 items.**

## Importing the Package

```zenscript
import mods.immersiveintelligence.AmmunitionWorkshop;
```

## Adding Recipes

### Parameters

| Type                                                | Name            | Required |
|-----------------------------------------------------|-----------------|----------|
| [IIngredient](/Vanilla/Variable_Types/IIngredient/) | Core Input      | Yes      |
| [IIngredient](/Vanilla/Variable_Types/IIngredient/) | Casing Input    | Yes      |
| IAmmunitionWorkshopFunction                         | Output Function | Yes      |
| int                                                 | Energy          | Yes      |
| int                                                 | Time            | Yes      |

### Syntax

```zenscript
mods.immersiveintelligence.AmmunitionWorkshop.addRecipe(IIngredient coreInput, IIngredient casingInput, IAmmunitionWorkshopFunction fun, int energy, int time);
```

### IAmmunitionWorkshopFunction

> IAmmunitionWorkshopFunction is a function used to determine the item outputted by a recipe based on ingredients.    

Reviewer: ```suggestion
IAmmunitionWorkshopFunction is a function used to determine the output item of a recipe based on its ingredients.    
```

> It can be used i.e. to get core NBT tag or merge core and casing's NBT tags.

Reviewer: ```suggestion
For example, It can be used to get the core NBT tag or merge the core and casing's NBT tags.
```
